### PR TITLE
Forms task

### PR DESCRIPTION
**Pull request by volha1**

### CODE REVIEW

File: src/components/Forms/MovieForm.tsx [hunk 1]
```diff
@@ -1,109 +1,206 @@
...
-            </label>
-            <div className="overSelect" />
-          </div>
-          {genresShown && (
-            <div className="genres">
+    <Formik
+      initialValues={movie}
+      enableReinitialize
+      validationSchema={Yup.object().shape({
+        genres: Yup.array().min(1, 'This field is required'),
```
Comment (YuryAniskou) [inline src/components/Forms/MovieForm.tsx:69]: Why do you need so many validation function if you install `Yup`. You can write validation rules here

File: src/components/Forms/MovieForm.tsx [hunk 2]
```diff
@@ -1,109 +1,206 @@
...
+            <div className="row">
+              <label>
+                Title
+                <br />
+                <Field
+                  type="text"
+                  name="title"
+                  className={classNames('left-field', { 'error-field': errors.title && touched.title })}
+                  value={values.title}
+                  validate={validateRequiredField}
```
Comment (YuryAniskou) [inline src/components/Forms/MovieForm.tsx:90]: Use Yup to validate fields

File: src/components/Forms/MovieForm.tsx [hunk 3]
```diff
@@ -1,109 +1,206 @@
...
-                Comedy
+                Movie url
+                <br />
+                <Field
+                  type="url"
+                  name="imgPath"
+                  className={classNames('left-field', { 'error-field': errors.imgPath && touched.imgPath })}
+                  placeholder="https://"
+                  value={values.imgPath}
+                  validate={validateImgPath}
```
Comment (YuryAniskou) [inline src/components/Forms/MovieForm.tsx:113]: Use Yup to validate fields

File: src/components/Forms/MovieForm.tsx [hunk 4]
```diff
@@ -1,109 +1,206 @@
...
-                Horror
+                Rating
+                <br />
+                <Field
+                  type="text"
+                  name="voteAverage"
+                  className={classNames('right-field', { 'error-field': errors.voteAverage && touched.voteAverage })}
+                  placeholder="7.8"
+                  value={values.voteAverage}
+                  validate={validateVoteAverage}
```
Comment (YuryAniskou) [inline src/components/Forms/MovieForm.tsx:127]: Use Yup to validate fields

File: src/components/GenreFilter/GenreFilter.tsx [hunk 5]
```diff
@@ -1,26 +1,37 @@
...
-const GenreFilter = (): ReactElement => {
+type ParamsProps = { filter: string | undefined; sortOrder: string; sortBy: string };
+type GenreFilterProps = {
+  onFilter: Dispatch<SetStateAction<ParamsProps>>;
+  params: ParamsProps;
+  genres: string[];
+};
+
+const GenreFilter = ({ onFilter, params, genres }: GenreFilterProps): ReactElement => {
+  const genreParam = params.filter === undefined ? genres[0] : params.filter;
```
Comment (YuryAniskou) [inline src/components/GenreFilter/GenreFilter.tsx:13]: You can write simpler. `params.filter || genres[0]` or `params.filter ?? genres[0]`

File: src/components/Messages/DeleteMovieMessage.tsx [hunk 6]
```diff
@@ -1,19 +1,31 @@
...
+  }, [movieIdForDeletion]);
+
   return (
     <div className="modal-window">
       <CloseBtn onClose={onClose} />
       <h1 className="title">Delete movie</h1>
       <p>Are you sure you want to delete this movie?</p>
       <div className="btn-wrapper">
-        <button type="button" className="btn" onClick={onClose}>
+        <button type="button" className="btn" onClick={handlehandleClick}>
```
Comment (YuryAniskou) [inline src/components/Messages/DeleteMovieMessage.tsx:28]: duplicate `handle` in `handlehandleClick`

File: src/components/Forms/MovieForm.tsx (hunk 1 above)
Comment (volha1) [reply src/components/Forms/MovieForm.tsx:69]: Fixed

File: src/components/Forms/MovieForm.tsx (hunk 2 above)
Comment (volha1) [reply src/components/Forms/MovieForm.tsx:90]: Fixed

File: src/components/Forms/MovieForm.tsx (hunk 3 above)
Comment (volha1) [reply src/components/Forms/MovieForm.tsx:113]: Fixed

File: src/components/Forms/MovieForm.tsx (hunk 4 above)
Comment (volha1) [reply src/components/Forms/MovieForm.tsx:127]: Fixed

File: src/components/GenreFilter/GenreFilter.tsx (hunk 5 above)
Comment (volha1) [reply src/components/GenreFilter/GenreFilter.tsx:13]: Fixed

File: src/components/Messages/DeleteMovieMessage.tsx (hunk 6 above)
Comment (volha1) [reply src/components/Messages/DeleteMovieMessage.tsx:28]: Fixed

**Pull request merged**